pkg/eks/tasks.go

```diff
@@ -247,15 +247,19 @@ func (c *ClusterProvider) ClusterTasksForNodeGroups(cfg *api.ClusterConfig, inst
 		IsSubTask: false,
 	}
 	var haveNeuronInstanceType bool
-	for _, ng := range cfg.NodeGroups {
-		haveNeuronInstanceType = haveNeuronInstanceType || api.HasInstanceType(ng, utils.IsInferentiaInstanceType)
-	}
 	var haveNvidiaInstanceType bool
 	for _, ng := range cfg.NodeGroups {
+		haveNeuronInstanceType = haveNeuronInstanceType || api.HasInstanceType(ng, utils.IsInferentiaInstanceType)
 		haveNvidiaInstanceType = haveNvidiaInstanceType || api.HasInstanceType(ng, func(t string) bool {
 			return utils.IsGPUInstanceType(t) && !utils.IsInferentiaInstanceType(t)
 		})
 	}
+	for _, ng := range cfg.ManagedNodeGroups {
+		haveNeuronInstanceType = haveNeuronInstanceType || api.HasInstanceTypeManaged(ng, utils.IsInferentiaInstanceType)
+		haveNvidiaInstanceType = haveNvidiaInstanceType || api.HasInstanceTypeManaged(ng, func(t string) bool {
+			return utils.IsGPUInstanceType(t) && !utils.IsInferentiaInstanceType(t)
```

Comment:
The body of this function is identical to the one used above. We can assign it to a var and reuse it.

Reply:
done